rest/config.go

```diff
@@ -842,6 +842,11 @@ func (sc *StartupConfig) validate() (errorMessages error) {
 			errorMessages = multierror.Append(errorMessages, fmt.Errorf("Cannot use supplied TLS key or cert when api.https.use_tls_client is false"))
 		}
 	}
+
+	if !base.IsEnterpriseEdition() && sc.API.EnableAdminAuthenticationPermissionsCheck != nil && *sc.API.EnableAdminAuthenticationPermissionsCheck {
```

Comment:
Could the check for enterprise edition be in an outer if statement then the config value check be nested so it is easy to add enterprise config value checks in the future without doing a redundant enterprise edition check?
```suggestion
        if !base.IsEnterpriseEdition() {
	        if sc.API.EnableAdminAuthenticationPermissionsCheck != nil && *sc.API.EnableAdminAuthenticationPermissionsCheck {
```

Reply:
Didn't bother doing this as this is the only check for EE in this path. If someone else wants to use it its fairly easy to split it out.